### PR TITLE
Don't install a preexec_fn by default

### PR DESCRIPTION
**Pull request by tiran**
ipautil.run() now only installs a preexec_fn when it is actually needed.
This addresses a compatibility issue with mod_wsgi subinterpreters under
Python 3.8.

See: https://bugs.python.org/issue37951
Fixes: https://bugzilla.redhat.com/show_bug.cgi?id=1759290
Signed-off-by: Christian Heimes <cheimes@redhat.com>

### CODE REVIEW

File: ipapython/ipautil.py [hunk 1]
```diff
@@ -515,14 +515,18 @@ def run(args, stdin=None, raiseonerr=True, nolog=(), env=None,
...
-            os.setregid(pent.pw_gid, pent.pw_gid)
-            os.setreuid(pent.pw_uid, pent.pw_uid)
-
-        if umask:
-            os.umask(umask)
+    if runas is not None or umask is not None:
+        # preexec function is not supported in WSGI environment
+        def preexec_fn():
+            if runas is not None:
+                os.setgroups(suplementary_gids)
```
Comment (gpshead) [inline ipapython/ipautil.py]: also worth noting: in Python 3.9 the subprocess module gains support for doing this without preexec_fn:
  https://bugs.python.org/issue36046

i don't believe we have umask support yet, but that one is easier to work around via a child process shell that calls umask before exec'ing the target.  For anything you find a preexec_fn being used for, feel free to file subprocess feature requests with upstream CPython.

**Comment by abbra** [comment]
LGTM. Please file requests for adding umask support to subprocess for future Python versions.

**Comment by gpshead** [comment]
https://bugs.python.org/issue38417 filed to track adding umask support in the future.

**Comment by tiran** [comment]
master:

* 0b8c81a5bc194ffac7ea713930d6b922d10285d4 Don't install a preexec_fn by default

**Pull request closed without merging**